### ISSUE TITLE
Update to latest version of react-reconciler

### ISSUE THREAD

**Issue by wearhere** [Yomguithereal/react-blessed#83]
Even though the README says that this library will work with React's latest v16.x.x, it does not seem that we can use [the new lifecycles and context APIs](https://reactjs.org/blog/2018/03/29/react-v-16-3.html) introduced in React v16.3 like the . I think this is because this library is using an outdated version of `react-reconciler`.

**Comment by Yomguithereal** [comment]
You are right. Unfortunately I don’t have much time right now. I will of course review and probably accept a PR fixing this issue :)
